pkg/networkservice/common/switchcase/default.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package switchcase
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+)
+
+// Condition is a type for swich.*Case condition
+type Condition = func(context.Context, *networkservice.Connection) bool
```

Comment:
Why do we need ctx?

```suggestion
type Condition = func(*networkservice.Connection) bool
```

Reply:
Hmm... I'd be curious the thinking here as well.  I could imagine potentially needing the context to get metadata, or to log etc... @Bolodya1997 did you have other thoughts as well?

Reply:
Yes, please take a look at this - https://github.com/networkservicemesh/sdk/pull/1047.

Reply:
Ah yes... I had not forseen that specific example... but something like that seemed like a likely need.  I'm good with ctx here.